### PR TITLE
Migrate building SwiftSyntax to swift_build_support

### PR DESCRIPTION
**Pull request by ahoppen**
This migrates the build infrastructure of SwiftSyntax to `swift_build_support`. It goes hand-in-hand with https://github.com/apple/swift-syntax/pull/160.

### CODE REVIEW

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/160

@swift-ci Please smoke test

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/160

@swift-ci Please smoke test

**Comment by ahoppen** [comment]
preset=buildbot_swiftsyntax_macos

@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/160

preset=buildbot_swiftsyntax_macos

@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
Great! The preset caught the generated files in SwiftSyntax being out-of-date. Let’s see if it passes now that the files have been updated.

https://github.com/apple/swift-syntax/pull/160

preset=buildbot_swiftsyntax_macos

@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
CI didn’t pick up the changes in the SwiftSyntax repo yet. Let’s fire it up again.

preset=buildbot_swiftsyntax_macos

@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
CI didn’t pick up the changes in the SwiftSyntax repo yet. Let’s fire it up again.

preset=buildbot_swiftsyntax_macos
@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/160
preset=buildbot_swiftsyntax_macos
@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/160
@swift-ci Please test

**Comment by ahoppen** [comment]
SwiftSyntax preset passed. 

Let’s try if SouceKit-LSP still builds since I touched its build script as well.

https://github.com/apple/swift-syntax/pull/160
preset=buildbot_sourcekitlsp_macos
@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
Building SourceKit-LSP also passed. https://ci.swift.org/job/swift-PR-osx-preset/60/

Let’s finally test the stress tester.

https://github.com/apple/swift-syntax/pull/160
preset= buildbot_skstresstester_macos
@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
Mistakingly added a space after `preset=` in the above comment. Let’s run again without the space.

https://github.com/apple/swift-syntax/pull/160
preset=buildbot_skstresstester_macos
@swift-ci Please test with preset macOS Platform

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/160
preset=buildbot_skstresstester_macos
@swift-ci Please test with preset macOS Platform

File: utils/build-script-impl [hunk 1]
```diff
@@ -101,8 +97,6 @@ KNOWN_SETTINGS=(
     installable-package         ""               "the path to the archive of the installation directory"
     test-installable-package    ""               "whether to run post-packaging tests on the produced package"
     reconfigure                 ""               "force a CMake configuration run even if CMakeCache.txt already exists"
-    build-libparser-only        ""               "only build libSwiftSyntaxParser"
```
Comment (benlangmuir) [inline utils/build-script-impl]: Why are we removing `--build-libparser-only`?  Isn't is still useful to be able to build the parser library without building the rest of the compiler?  CC @akyrtzi

Comment (akyrtzi) [reply utils/build-script-impl]: This has been superseded by running `utils/build-parser-lib`

**Review by benlangmuir — APPROVED** [review]

**Comment by aciidb0mb3r** [comment]
Cool! Is this read to land?

**Pull request merged**